dev-support/design-docs/HBASE-24289-Heterogeneous Storage for Date Tiered Compaction.md

```diff
@@ -0,0 +1,122 @@
+  policies (in HDFS) for data in different time windows.
++ DTCP designs different windows, and we can classify the windows according to
+  the timestamps of the windows. For example: HOT window, WARM window, COLD window.
++ DTCP divides storefiles into different windows, and performs minor Compaction within
+  a time window. The storefile generated by Compaction will use the storage strategy of
+  this window. For example, if a window is a HOT window, the storefile generated by compaction
+  can be stored on the SSD. There are already WAL and the entire CF support storage policy
+  (HBASE-12848, HBASE-14061), our goal is to achieve cold and hot separation in one CF or
+  a region, using different storage policies.
+
+## Definition of hot and cold data
+
+Usually the data of the last 3 days can be defined as `HOT data`, hot age = 3 days.
+ If the timestamp of the data is > (timestamp now - hot age), we think the data is hot data.
+ Warm age, cold age can be defined in the same way. Only one type of data is allowed.
+ ```
+  if timestamp >  (now - hot age) , HOT data
+  else if timestamp >  (now - warm age), WARM data
+  else if timestamp >  (now - cold age), COLD data
+  else  default, COLD data
+```
+
+## Time window
+When given a time now, it is the time when the compaction occurs. Each window and the size of
+ the window are automatically calculated by DTCP, and the window boundary is rounded according
+ to the base size.
+Assuming that the base window size is 1 hour, and each tier has 3 windows, the current time is
+ between 12:00 and 13:00. We have defined three types of winow (`HOT, WARM, COLD`). The type of
+ winodw is determined by the timestamp at the beginning of the window and the timestamp now.
+As shown in the figure 1 below, the type of each window can be determined by the age range
+ (hot / warm / cold) where (now - window.startTimestamp) falls. Cold age can not need to be set,
+ the default Long.MAX, meaning that the window with a very early time stamp belongs to the
+ cold window.
+![figure 1](https://raw.githubusercontent.com/pengmq1/images/master/F1-HDTCP.png "figure 1")
+
+## Example configuration
+
+| Configuration Key | value | Note |
+|:---|:---:|:---|
+|hbase.hstore.compaction.date.tiered.storage.policy.enable|true|if or not use storage policy for window. Default is false|
+|hbase.hstore.compaction.date.tiered.hot.window.age.millis|3600000|hot data age
+|hbase.hstore.compaction.date.tiered.hot.window.storage.policy|ALL_SSD|hot data storage policy, Corresponding HDFS storage policy
+|hbase.hstore.compaction.date.tiered.warm.window.age.millis|20600000||
+|hbase.hstore.compaction.date.tiered.warm.window.storage.policy|ONE_SSD||
+|hbase.hstore.compaction.date.tiered.cold.window.age.millis|Long.MAX||
+|hbase.hstore.compaction.date.tiered.cold.window.storage.policy|HOT||
+
+The original date tiered compaction related configuration has the same meaning and maintains
```

Comment:
What will happen if the CF config storage policy and enable this feature too?

Reply:
If `hbase.hstore.compaction.date.tiered.storage.policy.enable` is true, this will override CF config storage policy, and `hbase.hstore.block.storage.policy` does not work. Because storefile must belong to **one window** and will use window storage policy